core/eval.go

```diff
@@ -2469,6 +2469,34 @@ func evalHSET(args []string, store *Store) []byte {
 	return Encode(numKeys, false)
 }
 
+func evalHGETALL(args []string, store *Store) []byte {
+	if len(args) != 1 {
+		return diceerrors.NewErrArity("HGETALL")
+	}
+
+	key := args[0]
+
+	obj := store.Get(key)
+
+	var hashMap HashMap
+	var results []string
+
+	if obj != nil {
+		switch currentVal := obj.Value.(type) {
+		case HashMap:
+			hashMap = currentVal
+		default:
+			return diceerrors.NewErrWithMessage(diceerrors.WrongTypeErr)
+		}
+	}
```

Comment:
Let's use assertTypeEncoding for this check, and please do the same for existing evalHSET method.

Reply:
Done